src/containers/SidebarContainer.js

```diff
@@ -0,0 +1,64 @@
+import Sidebar from '../components/Sidebar';
+import { connect } from 'react-redux';
+import { ARTICLE, VIDEO, LIVE } from '../constants/ArticleTypes';
+import React from 'react';
+
+const mapStateToProps = ({articles, articleGroups}) => {
+  const content = {};
+  const contentTypes = [ARTICLE, VIDEO, LIVE];
+
+  // Find all used article IDs by going through articleGroups
+  const usedArticleIds = Object.keys(articleGroups)
+    .map( groupKey => articleGroups[groupKey].articles )
+    .reduce( (list, articleKeys) => list.concat(articleKeys) ,[]);
+
+  console.log("Used article ids", usedArticleIds);
+
+  // Get un-used article IDs
+  const unUsedArticleKeys = Object.keys(articles)
+    .filter(key => usedArticleIds.indexOf(parseInt(key, 10)) === -1);
+
+  console.log("Un-used article keys ", unUsedArticleKeys);
+
+  // Categorize by type
+  contentTypes.forEach( type => {
+    content[type] = unUsedArticleKeys
+      .filter(key => articles[key].type === type)
+      .map(key => articles[key]);
+  });
+
+  return {
+    contentTypes,
+    content
+  };
+}
+
+const mapDispatchToProps = () => {
+  return {
+    toggleSidebar: () => {
+      const sidebarClasses = document.getElementById("sidebar").classList;
```

Comment:
Not a fan of depending on DOM classes like this.

I think it's better to add a React state to the component which is toggled instead.
Something like 
```
toggle() {
  this.setState({
    active: !this.state.active
  });
}
```

And then this.state.active is sent to the Sidebar component as a prop.